index.js

```diff
@@ -1,15 +1,29 @@
 'use strict'
+/**
+ * @typedef {import('unified').FrozenProcessor} FrozenProcessor
+ * @typedef {import('unified').RunCallback} RunCallback
+ * @typedef {import('unified').Settings} Settings
+ * @typedef {import('unified').Transformer} Transformer
+ * @typedef {import('unist').Node} Node
+ */
 
 var hast2mdast = require('hast-util-to-mdast')
 
 module.exports = attacher
 
-// Attacher.
-// If a destination is given, runs the destination with the new mdast tree
-// (bridge-mode).
-// Without destination, returns the mdast tree: further plugins run on that tree
-// (mutate-mode).
+/**
+ * Attacher.
+ *
+ * If a destination is given, runs the destination with the new mdast
+ * tree (bridge-mode). Without destination, returns the mdast tree: further
+ * plugins run on that tree (mutate-mode).
+ *
+ * @param {FrozenProcessor} destination
+ * @param {Settings} options
```

Comment:
this is trying to represent an overload

```ts
attacher(destination: FrozenProcessor, options: Settings)
attacher(options: Settings)
attacher()
```

It's a bit easier to do in TS than JSDoc.
an imperfect version of this could be:
```suggestion
 * @param {FrozenProcessor | Settings} [destination]
 * @param {Settings} [options]
```

But IIRC @wooorm has a trick for getting proper overloads in working JSDoc

Reply:
Here’s an example of how I do overloads. If you want to learn, go for it! If you’d rather thing it’s too much, feel free to go with Christian’s suggestion.

https://github.com/micromark/micromark/blob/0b0e68dcda98a962afddb7b0d59b719d853f8456/packages/micromark/dev/index.js#L12-L42

The first comment block is just docs attached to parameters. In my editor (Atom), I don’t get descriptions of parameters from the last block.

The second block is the different signatures a function has. The first and second blocks are what is exposed to users of the function.

The third block is what’s used internally. You can use Christian’s suggestion there.

Note that the parens are a bit weird: prettier and TS have some weird handling of them, so make sure to exactly stick with the way it’s written in the link I gave!

Reply:
Thanks! Tried something out, but I did have to edit the JS code a bit to make it work and I had to cast. Let me know if I got it right.